client/src/components/common/boost/BoostConfigList.tsx

```diff
@@ -23,13 +30,25 @@ export default function BoostConfigList({
   onSelectConfig,
   onDeleteConfig,
 }: BoostConfigListProps) {
-  const handleSelect = (configName: string) => {
-    if (configName !== config.active) onSelectConfig(configName);
+  const handleSelect = (configFileName: string) => {
+    if (configFileName !== config.active?.fileName)
+      onSelectConfig(configFileName);
   };
   const handleDelete = (event: React.MouseEvent, configName: string) => {
     event.stopPropagation();
     onDeleteConfig(configName);
   };
+
+  const getOriginalFileName = (fileName: string) => {
+    let originalFileName = fileName;
+    fileNameSuffixes.forEach((suffix: string) => {
+      if (fileName.includes(suffix)) {
+        originalFileName = fileName.replace(suffix, '.json');
+      }
+    });
+    return originalFileName;
+  };
```

Comment:
This could then be generalised as a util function if the above suggestion was implemented.